### PR TITLE
Add validation from wrong period type for the exchange rates + other improv.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-4894

### What?

- Add validation from the wrong period type for the exchange rates.
- Small refactor or files_controller.
- Rename type to period_type for clarity.

### Why?
To improve error messages and error status.
Make the code cleaner.

### CODE REVIEW

File: spec/requests/api/v2/exchange_rates/files_controller_spec.rb [hunk 1]
```diff
@@ -79,15 +81,28 @@
...
+
+    context 'when requesting invalid period type' do
+      let(:period_type) { 'invalid_period_type' }
+      let(:format) { 'csv' }
+
+      before { get api_exchange_rates_file_path("#{period_type}_#{year}-#{month}", format:) }
+
+      it 'returns HTTP 400 bad request' do
+        expect(response).to have_http_status(:bad_request)
+      end
```
Comment (alexdesi) [inline spec/requests/api/v2/exchange_rates/files_controller_spec.rb:105]: This is the new behaviour that has been added.
When the period_type is wrong, it returns bad request 400.

File: app/controllers/api/v2/exchange_rates/files_controller.rb [hunk 2]
```diff
@@ -3,7 +3,9 @@ module V2
     module ExchangeRates
       class FilesController < ApiController
         def show
-          filename = ExchangeRateFile.filename_for_download(type, format, year, month)
+          validate_params!
+
+          filename = ExchangeRateFile.filename_for_download(period_type, format, year, month)
```
Comment (alexdesi) [inline app/controllers/api/v2/exchange_rates/files_controller.rb:8]: renamed `type` -> `period_type`, for clarity.

File: app/controllers/api/v2/exchange_rates/files_controller.rb [hunk 3]
```diff
@@ -48,12 +50,19 @@ def file_data
...
-            type:,
+            type: period_type,
             period_year: year,
             period_month: month,
             format: format.to_s,
           ).take
         end
+
+        def validate_params!
+          if period_type.nil? # && ExchangeRateFile::APPLICABLE_TYPES.include?(period_type)
```
Comment (alexdesi) [inline app/controllers/api/v2/exchange_rates/files_controller.rb:61]: need to remove this comment.

File: app/models/exchange_rate_file.rb [hunk 4]
```diff
@@ -25,10 +25,6 @@ def filename
   end
 
   dataset_module do
-    def applicable_types
-      where(type: APPLICABLE_TYPES)
-    end
-
```
Comment (alexdesi) [inline app/models/exchange_rate_file.rb:31]: this is not used

**Comment by alexdesi** [comment]
Clear error message when period type is invalid. 
This is from local (on prod it trigger 400 bad request): 
image

**Review by HWallenberg — APPROVED** [review]
lgtm

**Review by shjohnson — APPROVED** [review]
LGTM

**Pull request closed without merging**